### PR TITLE
fix: sanitize primary domain for orgs

### LINKED ISSUE #5579: [Bug]: Primary domain for organizations not being sanitized
### Preflight Checklist

- [X] I could not find a solution in the documentation, the existing issues or discussions
- [X] I have joined the [ZITADEL chat](https://zitadel.com/chat)

### Environment

ZITADEL Cloud

### Describe the bug

Primary domain for organizations should be sanitized before being set.

### To reproduce

1. Log in to a Zitadel cloud instance
2. Create a new organization with non-alphanumeric characters.
3. Check the Primary Domain tab for the new organization
  - e.g. If I make a new organization named `test organization 3 ()`, the domain for it will be `test-organization-3-().my-zitadel-instance.zitadel.cloud`

### Screenshots

image

### Expected behavior

The domain should be sanitized before actually being deployed
e.g. `test-organization-3.my-zitadel-instance.zitadel.cloud`

### Version

2.22.3

### Operating System

N/A

### Relevant Configuration

_No response_

### Additional Context

Please feel free to log-in to my zitadel instance `instance1-4h3lvy.zitadel.cloud` to validate this issue.

### PR DESCRIPTION
**Pull request by doncicuto**
As @sanvit stated: "Primary domain for organizations should be sanitized before being set. [...] If I make a new organization named test organization 3 (), the domain for it will be test-organization-3-().my-zitadel-instance.zitadel.cloud" 

This PR implements a funcion to sanitize the primary domain that uses the name of the organization, removing invalid characters and checking the number of characters. [https://www.ietf.org/rfc/rfc1035.txt](https://www.ietf.org/rfc/rfc1035.txt) and [https://www.nic.ad.jp/timeline/en/20th/appendix1.html](https://www.nic.ad.jp/timeline/en/20th/appendix1.html) is used as reference. 

In the following screenshot we have two orgs with a domain name. The first one shows the described issue, the later shows how a right domain name is set based on the org name.

sanitize_domain

Should close #5579

### Definition of Ready

- [X] I am happy with the code
- [X] Short description of the feature/issue is added in the pr description
- [X] PR is linked to the corresponding user story
- [X] Acceptance criteria are met
- [ ] All open todos and follow ups are defined in a new ticket and justified
- [ ] Deviations from the acceptance criteria and design are agreed with the PO and documented.
- [X] No debug or dead code
- [X] My code has no repetitions
- [X] Critical parts are tested automatically
- [ ] Where possible E2E tests are implemented
- [ ] Documentation/examples are up-to-date
- [ ] All non-functional requirements are met
- [X] Functionality of the acceptance criteria is checked manually on the dev system.

### CODE REVIEW

**Comment by hifabienne** [comment]
Thanks for you pr.
We will have a look at it.

File: internal/utils/utils.go [hunk 1]
```diff
@@ -0,0 +1,44 @@
+package utils
+
+import (
+	"regexp"
+	"strings"
+)
+
+func SanitizeDomain(name, suffix string) string {
```
Comment (stebenz) [inline internal/utils/utils.go:8]: Could you please add unit tests for the different variants, to limit the impact don't the code which uses this function?

File: internal/org/model/org.go [hunk 2]
```diff
@@ -47,7 +46,8 @@ func (o *Org) GetPrimaryDomain() *OrgDomain {
 }
 
 func (o *Org) nameForDomain(iamDomain string) string {
```
Comment (stebenz) [inline internal/org/model/org.go:48]: Can be deleted as only used in AddIAMDomain a few lines down and then can point to NewIAMDomainName.
A separate package utils can be deleted and included in internal/domain, as it is logic related to the domains.

File: internal/command/org_test.go [hunk 3]
```diff
@@ -456,6 +456,158 @@ func TestCommandSide_AddOrg(t *testing.T) {
 				},
 			},
 		},
+		{
+			name: "add org (remove invalid characters), no error",
```
Comment (stebenz) [inline internal/command/org_test.go:460]: "no error" can be ignored, as sanitization will never result in an error. There are a lot more possible combinations but that can be handled in unit tests specific for the sanitization.

**Review by stebenz — CHANGES_REQUESTED** [review]

File: internal/utils/utils.go (hunk 1 above)
Comment (doncicuto) [reply internal/utils/utils.go:8]: Sure! I'll prepare them this weekend.

File: internal/org/model/org.go (hunk 2 above)
Comment (doncicuto) [reply internal/org/model/org.go:48]: Absolutely

File: internal/utils/utils.go (hunk 1 above)
Comment (doncicuto) [reply internal/utils/utils.go]: Done!

File: internal/org/model/org.go (hunk 2 above)
Comment (doncicuto) [reply internal/org/model/org.go]: Done!

File: internal/command/org_test.go (hunk 3 above)
Comment (doncicuto) [reply internal/command/org_test.go]: Done!

**Comment by doncicuto** [comment]
Thanks @stebenz for your review, if you need me to made additional changes let me know

**Review by stebenz — APPROVED** [review]
Thanks @doncicuto

**Pull request merged**